--- src/scheduler_functions.c
+++ src/scheduler_functions.c
@@ -8,7 +8,7 @@
     if (left->arrival_time != right->arrival_time)
         return (left->arrival_time < right->arrival_time);
     
-    return (left->c_id < right->c_id);
+    return (left->c_id > right->c_id);
 }
 
 int edf(void *e1, void *e2)
@@ -19,7 +19,10 @@
     if (left->last_compilation != right->last_compilation)
         return (left->last_compilation < right->last_compilation);
     
-    return (left->c_id < right->c_id);
+    if (left->compilation_count != right->compilation_count)
+        return (left->compilation_count > right->compilation_count);
+
+    return (left->c_id > right->c_id);
 }
